### PR TITLE
Added filter to show free and paid events

### LINKED ISSUE #2901: [FEATURE] Events page - add a cost identifier
### Description

This feature is to add a legend item for cost to events so that folks can easily identify free vs paid events.

To keep things simple proposing only
Free Event (possible icon tbd or simply put Free as text)
Paid Event (possible icon FaMoneyBill)

Consideration: If the event has a free and a paid portion, do we need a way to show that?

We could get more granular with this in the future if people find it more useful to know precise cost without clicking through to the event

### Screenshots

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by shyam0705**
## Fixes Issue
Closes #2901 

## Changes proposed 

1)Added filter to show all free and paid events.
2)Modified documentation to suggest adding the `startingFrom` field in events 

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [x] My change requires changes to the documentation.
- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Screenshots

UI:-
UI
Modified events for testing purpose:-
e1
e2

## Note to reviewers

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/3110"></a>

### CODE REVIEW

File: pages/docs/how-to-guides/events.mdx [hunk 1]
```diff
@@ -19,11 +19,12 @@ All future events also appear on the Events page for the app.
 
 _If you need help on how to edit this file, please see the [Editing guide](/docs/how-to-guides/editing)_
 
-3. This `json` file will contain one object for the event and must have six fields, `isVirtual` and/or `isInPerson`, `color`, `name`, `description`, `date` and `url`, which will look like this:
+3. This `json` file will contain one object for the event and must have seven fields, `startingFrom`,`isVirtual`and/or`isInPerson`, `color`, `name`, `description`, `date`and`url`, which will look like this:
```
Comment (eddiejaoude) [inline pages/docs/how-to-guides/events.mdx:22]:
```suggestion
3. This `json` file will contain one object for the event and must have seven fields, `startingFrom`, `isVirtual` and/or `isInPerson`, `color`, `name`, `description`, `date`and`url`, which will look like this:
```

File: pages/docs/how-to-guides/events.mdx [hunk 2]
```diff
@@ -19,11 +19,12 @@ All future events also appear on the Events page for the app.
...
 
 <ClipboardCopy>
 ```js
 {
+  "startingFrom":0,
```
Comment (eddiejaoude) [inline pages/docs/how-to-guides/events.mdx:27]:
```suggestion
  "startingFrom": 0,
```

**Review by eddiejaoude — CHANGES_REQUESTED** [review]
Looks good, I think the field should be more descriptive, something like ...

```js
price: {
   startingFrom: 0
}
```

**Comment by shyam0705** [comment]
> Looks good, I think the field should be more descriptive, something like ...
> 
> ```js
> price: {
>    startingFrom: 0
> }
> ```

Thanks @eddiejaoude, I will make changes soon.

**Comment by eddiejaoude** [comment]
Thanks for making the changes 👍  During testing, I got this error on the user profile page...

Screenshot 2023-01-12 at 10 35 15

**Comment by eddiejaoude** [comment]
Also please update `main` into this branch to reduce any future conflicts

**Comment by eddiejaoude** [comment]
> Can you check now? I'm sorry; I completely missed adding free and paid categories in the user event tab.

no problem, these things happen to everyone

These changes are going to have challenges with another PR https://github.com/EddieHubCommunity/LinkFree/pull/3125

I think the extra filter is a good idea, but not sure how to display it so it doesn't take up too much space on the page 🤔

**Comment by shyam0705** [comment]
> > Can you check now? I'm sorry; I completely missed adding free and paid categories in the user event tab.
> 
> no problem, these things happen to everyone
> 
> These changes are going to have challenges with another PR #3125
> 
> I think the extra filter is a good idea, but not sure how to display it, so it doesn't take up too much space on the page 🤔

I guess we can modify @Aadarsh805  design to include the following and previous icons. We can have a design like pagination.

**Comment by shyam0705** [comment]
> @eddiejaoude and @shyam0705, I noticed this has stalled, can we pick it back up? I saw the comment about the filters but we could break that part out into a separate issue. wdyt?

@amandamartin-dev  sorry, I was busy due to some personal reasons last week. will pick this back. I am thinking about creating a separate issue to display filters as we add more filters in future.

**Pull request closed without merging**